controllers/Controller.js

```diff
@@ -0,0 +1,31 @@
+const {getAllTopics,articleById} = require('../models/Models.js')
+
+exports.getTopics =(req,res,next)=>{
+    getAllTopics()
+.then((topics)=>{
+        
+       res.status(200).send(topics) 
+})
+.catch(err => {
+    next(err);   
+})
+
+}
+
+exports.getArticleId = (req, res, next) => {
+    const { article_id } = req.params;
+    
+    if (isNaN(parseInt(article_id))) {
+        return res.status(400).send({ error: 'Invalid article ID format' });
+    }
+     articleById(article_id)
+     .then((article)=>{
```

Comment:
For a valid but non-existent article, i would recommend rejecting the promise in the model, which would then pipe the error into our error handling middleware